From 2bd4d99a32461dad09deb0b0db933b4a650ad1ef Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Fri, 1 Aug 2025 07:44:42 +0200
Subject: [PATCH 17/55] base-files: sysupgrade: always assume -n if -f is not
 provided

Changes sysupgrade default behaviour to not preserve configuration unless explicitly requested with -f flag, avoiding config carry-over during firmware updates in managed environments.

Signed-off-by: John Crispin <john@phrozen.org>
---
 package/base-files/files/sbin/sysupgrade | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/package/base-files/files/sbin/sysupgrade b/package/base-files/files/sbin/sysupgrade
index 611d8830bf..9bdbe9f625 100755
--- a/package/base-files/files/sbin/sysupgrade
+++ b/package/base-files/files/sbin/sysupgrade
@@ -29,7 +29,7 @@ export MTD_ARGS=""
 export MTD_CONFIG_ARGS=""
 export INTERACTIVE=0
 export VERBOSE=1
-export SAVE_CONFIG=1
+export SAVE_CONFIG=0
 export IGNORE_MINOR_COMPAT=0
 export FORCE=0
 export CONFFILES=/tmp/sysupgrade.conffiles
@@ -389,9 +389,6 @@ if [ -n "$CONF_IMAGE" ]; then
 	esac
 	get_image "$CONF_IMAGE" "cat" > "$CONF_TAR"
 	export SAVE_CONFIG=1
-elif ask_bool $SAVE_CONFIG "Keep config files over reflash"; then
-	[ $TEST -eq 1 ] || create_backup_archive "$CONF_TAR" || exit
-	export SAVE_CONFIG=1
 else
 	[ $TEST -eq 1 ] || rm -f "$CONF_TAR"
 	export SAVE_CONFIG=0
-- 
2.34.1

